src/main/java/org/apache/commons/lang3/builder/TypedEqualsBuilder.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.lang3.builder;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+/**
+ * An extension of {@link EqualsBuilder} aimed to perform additionally the base objects and class equals checks.
+ *
+ * <p>It then offers the possibility to append field extractors.</p>
+ *
+ * <p>Typical use for the code is as follows:</p>
+ * <pre>
+ * public boolean equals(Object obj) {
+ *     return new TypedEqualsBuilder<>(this)
+ *         .appendBaseObject(obj)
+ *         .append(TestObject::getA)
+ *         .append(TestObject::getB)
+ *         .isEquals();
+ *     }
+ * </pre>
+ *
+ * @param <T> the type of the compared object.
+ *
+ * @since 3.14.0
+ */
+public class TypedEqualsBuilder<T> extends EqualsBuilder {
```

Comment:
This is a confusing class name IMO. What is and where is the Type? It's not a Java `Type` or `Class` apparently...

Reply:
It is the class type. its refering to the T class parameter which is supported by this Builder subtype.

Reply:
Yes it is a java Class. But maybe I should rename it differently or extend the EqualsBuilder and add some special appending.... I will try that...

Reply:
This equals builder is typed. We provide a type and then the field functions to build the equals method.
So the Type is a Java Class.

Reply:
> It is the class type. its refering to the T class parameter which is supported by this Builder subtype.

✅️ Thanks for pointing that out! 👍